spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -515,7 +538,15 @@ protected void doShutdown() {
 						consumer.basicCancel(true);
 						canceledConsumers.add(consumer);
 						consumerIterator.remove();
+						if (consumer.declaring) {
+							consumer.thread.interrupt();
```

Comment:
I think we can't do this. Because thread will be dead afterwards in the `Executor` and we might not be able to restart container afterward.